packages/test/test-end-to-end-tests/src/test/benchmark/DocumentMultipleDds.ts

```diff
@@ -0,0 +1,241 @@
+	private _mainContainer: IContainer | undefined;
+	private containerRuntime: ContainerRuntime | undefined;
+	private mainDataStore: TestDataObject | undefined;
+	private readonly dsCounts: number;
+	private readonly dsCountPerIteration: number;
+	private readonly _dataObjectFactory: DataObjectFactory<TestDataObject>;
+	public get dataObjectFactory() {
+		return this._dataObjectFactory;
+	}
+	private readonly runtimeFactory: ContainerRuntimeFactoryWithDefaultDataStore;
+
+	public get mainContainer(): IContainer | undefined {
+		return this._mainContainer;
+	}
+
+	public get logger(): ITelemetryLogger | undefined {
+		return this.props.logger;
+	}
+
+	private async ensureContainerConnectedWriteMode(container: IContainer): Promise<void> {
+		const resolveIfActive = (res: () => void) => {
+			if (container.deltaManager.active) {
+				res();
+			}
+		};
+		const containerConnectedHandler = (_clientId: string): void => {};
+
+		if (!container.deltaManager.active) {
+			await new Promise<void>((resolve) =>
+				container.on("connected", () => resolveIfActive(resolve)),
+			);
+			container.off("connected", containerConnectedHandler);
+		}
+	}
+
+	private async createDataStores() {
+		assert(
+			this._mainContainer !== undefined,
+			"Container should be initialized before creating data stores",
+		);
+		assert(
+			this.containerRuntime !== undefined,
+			"ContainerRuntime should be initialized before creating data stores",
+		);
+		assert(
+			this.mainDataStore !== undefined,
+			"mainDataStore should be initialized before creating data stores",
+		);
+		const totalIterations = this.dsCounts / this.dsCountPerIteration;
+		for (let i = 0; i < totalIterations; i++) {
+			for (let j = 0; j < this.dsCountPerIteration; j++) {
+				const dataStore = await this.dataObjectFactory.createInstance(
+					this.containerRuntime,
+				);
+				this.mainDataStore._root.set(`dataStore${j}`, dataStore.handle);
+			}
+			await this.waitForContainerSave(this._mainContainer);
+		}
+	}
+
+	private async waitForContainerSave(c: IContainer) {
+		if (!c.isDirty) {
+			return;
+		}
+		await new Promise<void>((resolve) => c.on("saved", () => resolve()));
+	}
+
+	/**
+	 * Creates a new DocumentCreator using configuration parameters.
+	 * @param props - Properties for initializing the Document Creator.
+	 * @param numberOfKeysInMap - Size of the document to be created 1=5Mb, 2=10Mb, etc.
+	 */
+	public constructor(private readonly props: IDocumentProps) {
+		this._dataObjectFactory = new DataObjectFactory(
+			"TestDataObject",
+			TestDataObject,
+			[SharedMap.getFactory(), SharedString.getFactory()],
+			[],
+		);
+		this.runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			this.dataObjectFactory,
+			[[this.dataObjectFactory.type, Promise.resolve(this.dataObjectFactory)]],
+			undefined,
+			undefined,
+			runtimeOptions,
+		);
+
+		switch (this.props.documentType) {
+			case "MediumDocumentMultipleDDSs":
```

Comment:
Actually, it creates 250 data stores not DDSes. There will be 250 data stores and 750 DDSes.
For the next test, its 500 data stores and 1500 DDSes.

The documentType name should be updated to reflect that.

Reply:
Maybe `MediumDocumentMultipleDataStores` and then add a comment in the type declaration to say each data store has 3 DDS?